### ISSUE TITLE
PyPI page claims compatibility with Python >=3.5

### ISSUE THREAD

**Issue by thetic** [ryanrhee/shellcheck-py#5]
at least 3.6 is needed

**Comment by asottile** [comment]
could you elaborate?

## python 2.7

```console
$ python --version
Python 2.7.17
$ python -m pip install shellcheck-py
Collecting shellcheck-py
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/ea/d6/e3d1fc934774d5f2d2fd76484baa3bbb4f3172aa9eccff1a6d5bcc405c90/shellcheck_py-0.7.0.1-py2.py3-none-manylinux1_x86_64.whl (2.3MB)
    100% |████████████████████████████████| 2.3MB 698kB/s 
Installing collected packages: shellcheck-py
Successfully installed shellcheck-py-0.7.0.1
You are using pip version 8.1.2, however version 20.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.7.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

## python3.5

```console
$ python --version
Python 3.5.9
$ python -mpip install shellcheck-py
Collecting shellcheck-py
  Using cached https://files.pythonhosted.org/packages/ea/d6/e3d1fc934774d5f2d2fd76484baa3bbb4f3172aa9eccff1a6d5bcc405c90/shellcheck_py-0.7.0.1-py2.py3-none-manylinux1_x86_64.whl
Installing collected packages: shellcheck-py
Successfully installed shellcheck-py-0.7.0.1
WARNING: You are using pip version 19.3.1; however, version 20.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.7.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

**Comment by asottile** [comment]
seems maybe due to https://github.com/pypa/pip/issues/7626 ?

**Comment by thetic** [comment]
This problem seems isolated to pipenv. I cannot reproduce natively or in a plain virtual environment.

```
$ python --version
Python 3.5.9

$ python -mpip install  shellcheck-py
Looking in indexes: <redacted internal URL>
Collecting shellcheck-py
  Downloading <redacted internal URL>
    ERROR: Command errored out with exit status 1:
     command: /home/<user>/.local/share/virtualenvs/mwe-py4r1y5y/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ri100t9b/shellcheck-py/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ri100t9b/shellcheck-py/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-ri100t9b/shellcheck-py/pip-egg-info
         cwd: /tmp/pip-install-ri100t9b/shellcheck-py/
    Complete output (6 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-ri100t9b/shellcheck-py/setup.py", line 31
        return f'shellcheck-v{SHELLCHECK_VERSION}.{get_arch()}'
                                                              ^
    SyntaxError: invalid syntax
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by asottile** [comment]
is the wheel available on your `<redacted internal URL>`?  it is correct that `shellcheck-py` requires python3.6+ to build from source, but it doesn't require that to install it

are you using pip 20.x? pip 20.x is currently broken for downloading `*-none-{arch}` wheels

**Comment by asottile** [comment]
this appears fixed with pip 20.0.2:

```console
$ pip install shellcheck-py
Collecting shellcheck-py
  Downloading shellcheck_py-0.7.0.1-py2.py3-none-manylinux1_x86_64.whl (2.3 MB)
     |████████████████████████████████| 2.3 MB 5.3 MB/s 
Installing collected packages: shellcheck-py
Successfully installed shellcheck-py-0.7.0.1
```
